src/inspector_socket.cc

```diff
@@ -192,7 +192,7 @@ static bool IsIPAddress(const std::string& host) {
     // Parse the IPv6 address to ensure it is syntactically valid.
     char ipv6_str[INET6_ADDRSTRLEN];
     std::copy(host.begin() + 1, host.end() - 1, ipv6_str);
-    ipv6_str[host.length()] = '\0';
+    ipv6_str[host.length() - 2] = '\0';
```

Comment:
A better approach is to use Ada parser and validate it is IPv6

Reply:
The `IsIPAddress` function is invoked by the [`IsAllowedHost`](https://github.com/nodejs/node/blob/main/src/inspector_socket.cc#L639-L643) function, which accepts a `host` parameter. Typically, the `host` represents an IP address that may include a port number. For instance, in the test case found [here](https://github.com/nodejs/node/blob/main/test/parallel/test-inspector-reported-host.js#L10), the `host` is solely an IP address.

Upon reviewing the Ada parser, it appears to be designed primarily for parsing URLs. So, I can't use it here like this: `ada::parse(host)`.

Is there a potential method to adapt the Ada parser for use with IP addresses?

Reply:
Since this is a bug fix, keeping the patch minimal is a good idea. Refactoring to use Ada or some entirely different validation instead does not need to happen in this PR.